stormkit/main_test.go

```diff
@@ -27,30 +33,318 @@ func TestConfig(t *testing.T) {
-	assert.Equal(t, expectedUseHTTPS, useHTTPS)
-	assert.Equal(t, expectedEngineAppID, engineAppID)
+	assert.Equal(t, expectedServer, globalConfig.Server)
+	assert.Equal(t, expectedBearerToken, globalConfig.BearerToken)
+	assert.Equal(t, time.Duration(expectedClientTimeout), globalConfig.ClientTimeout)
+	assert.Equal(t, expectedUseHTTPS, globalConfig.UseHTTPS)
+	assert.Equal(t, expectedEngineAppID, globalConfig.AppID)
 }
 
 // TestEngineAppID runs a sequence of manipulation on engineAppID
 func TestEngineAppID(t *testing.T) {
 	viperInit()
 
-	Config()
+	Config("")
 
-	assert.Equal(t, expectedEngineAppID, engineAppID)
+	assert.Equal(t, expectedEngineAppID, globalConfig.AppID)
 	assert.Equal(t, expectedEngineAppID, GetEngineAppID())
 
 	localEngineAppID := "test"
 
 	SetEngineAppID(localEngineAppID)
 
-	assert.Equal(t, localEngineAppID, engineAppID)
+	assert.Equal(t, localEngineAppID, globalConfig.AppID)
 	assert.Equal(t, localEngineAppID, GetEngineAppID())
 	assert.Equal(t, localEngineAppID, viper.Get(engineAppIDString))
 }
+
+func TestEngineAppIDConfigFile(t *testing.T) {
+	viperInit()
+	expectedAppID := "aaaa"
+	fi := localFileInfo{}
+	osStat = func(p string) (os.FileInfo, error) {
+		return &fi, nil
+	}
+	fi.IsDirVar = false
+
+	ioutilReadFile = func(p string) ([]byte, error) {
+		return []byte(`
+app:
+  - id: ` + expectedAppID), nil
+	}
+
+	re := captureOutput(func() {
+		Config(".")
+	})
+
+	assert.Equal(t, "", re)
+}
+
+func TestEngineAppIDConfigFileTwoApps(t *testing.T) {
+	viperInit()
+	expectedAppID := "aaaa"
+	fi := localFileInfo{}
+	osStat = func(p string) (os.FileInfo, error) {
+		return &fi, nil
+	}
+	fi.IsDirVar = false
+
+	ioutilReadFile = func(p string) ([]byte, error) {
+		return []byte(`
+app:
+  - id: ` + expectedAppID + `
+  - id: "ccccc"`), nil
+	}
+
+	re := captureOutput(func() {
+		Config(".")
+	})
+
+	assert.Equal(t, "there are many apps in the config file (using the first)\n", re)
+}
+
+func captureOutput(f func()) string {
+	reader, writer, err := os.Pipe()
+	if err != nil {
+		panic(err)
+	}
+	stdout := os.Stdout
+	stderr := os.Stderr
+	defer func() {
+		os.Stdout = stdout
+		os.Stderr = stderr
+		log.SetOutput(os.Stderr)
+	}()
+	os.Stdout = writer
+	os.Stderr = writer
+	log.SetOutput(writer)
+	out := make(chan string)
+	wg := new(sync.WaitGroup)
+	wg.Add(1)
+	go func() {
+		var buf bytes.Buffer
+		wg.Done()
+		io.Copy(&buf, reader)
+		out <- buf.String()
+	}()
+	wg.Wait()
+	f()
+	writer.Close()
+	return <-out
+}
+
+func TestGetStormkitConfigFilePathNotExits(t *testing.T) {
+	osStat = func(p string) (os.FileInfo, error) {
+		return nil, os.ErrNotExist
+	}
+
+	p, err := getStormkitConfigFilePath("")
+
+	assert.Empty(t, p)
+	assert.Equal(t, os.ErrNotExist, err)
+}
+
+type localFileInfo struct {
+	NameVar    string
+	SizeVar    int64
+	ModeVar    os.FileMode
+	ModTimeVar time.Time
+	IsDirVar   bool
+	SysVar     interface{}
+}
+
+func (fi *localFileInfo) Name() string {
```

Comment:
Because is an implementation of `os.FileInfo` ([GitHub](https://github.com/golang/go/blob/go1.15.4/src/os/types.go#L21), [`pkg.go.dev`](https://pkg.go.dev/os#FileInfo)), so are required all the methods, (the oder are public for easier testing.